### PR TITLE
Fix bug where job matrices with leading numbers generated duplicate job names

### LINKED ISSUE #2932: [matrix generator] Matrices with leading numbers cause duplicates to be pruned
As devops only supports [job matrix keys that start with a letter](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/phases?view=azure-devops&tabs=yaml#multi-job-configuration), it causes matrices with leading numbers to fail to generate properly (modified values and/or duplicates being dropped). Relevant code: https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/scripts/job-matrix/job-matrix-functions.ps1#L521

We should either:
1. Fail outright in these cases and force config authors to not use number values for the first matrix entry
2. Support some sort of alphanumeric prefix for leading numbers
3. Re-arrange the ordering (this breaks job display name determinism)
4. ?

Examples
```
{
    "matrix": {
        "foo": ["1", "2"],
        "bar": ["a", "b"]
    }
}
```
```
{
    "matrix": {
        "foo": [1, 2],
        "bar": ["a", "b"]
    }
}
```

Both will result in the following warning and an incomplete matrix like:

```
$ ./eng/common/scripts/job-matrix/Create-JobMatrix.ps1 -Selection all -ConfigPath ./test-matrix.json
WARNING: Found duplicate configurations for job "a". Multiple values may have been replaced with the same value.
WARNING: Found duplicate configurations for job "b". Multiple values may have been replaced with the same value.
{
  "a": {
    "foo": "1",
    "bar": "a"
  },
  "b": {
    "foo": "1",
    "bar": "b"
  }
}
```

However, switching the order around works fine, because the job names won't have leading numbers:

```
{
    "matrix": {
        "bar": ["a", "b"],
        "foo": ["1", "2"]
    }
}
```

Generates correctly with:

```
$ ./eng/common/scripts/job-matrix/Create-JobMatrix.ps1 -Selection all -ConfigPath ./test-matrix.json
{
  "a_1": {
    "bar": "a",
    "foo": "1"
  },
  "a_2": {
    "bar": "a",
    "foo": "2"
  },
  "b_1": {
    "bar": "b",
    "foo": "1"
  },
  "b_2": {
    "bar": "b",
    "foo": "2"
  }
}
```

### PR DESCRIPTION
**Pull request by benbp**
Fixes #2932

- Fix bug where job matrices with leading numbers generated duplicate job names
- Fail matrix generation when config path or import paths are not found

### CODE REVIEW

**Comment by benbp** [comment]
@JimSuplizio this was that weird issue we were seeing when I was demoing the matrix generator to you.

File: eng/common/scripts/job-matrix/job-matrix-functions.ps1
Comment (weshaggard) [inline eng/common/scripts/job-matrix/job-matrix-functions.ps1:522]: Are there any other special chars in this set?

**Review by weshaggard — APPROVED** [review]
Looks reasonable but please hold off until after release week to merge.

File: eng/common/scripts/job-matrix/job-matrix-functions.ps1
Comment (benbp) [reply eng/common/scripts/job-matrix/job-matrix-functions.ps1:522]: We could match on any non-alpha characters instead of just digits, which would be more flexible.

Comment (benbp) [reply eng/common/scripts/job-matrix/job-matrix-functions.ps1:522]: Actually, this is not necessary because the job name will already filter out non alphanumeric characters since they aren't supported at all by devops.

**Pull request merged**
